pkg/framework/test/apiserver_test.go

```diff
@@ -0,0 +1,51 @@
+package test_test
+
+import (
+	. "k8s.io/kubectl/pkg/framework/test"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+	"github.com/onsi/gomega/gexec"
+)
+
+var _ = Describe("Apiserver", func() {
+
+	Context("when given a path to a binary that runs for a long time", func() {
+		It("can start and stop that binary", func() {
+			pathToFakeAPIServer, err := gexec.Build("k8s.io/kubectl/pkg/framework/test/assets/fakeapiserver")
```

Comment:
When you say "the coupling with `main.go` below is kind of awkward", we think you're primarily referring to the duplication between the arguments in [test/apiserver.go](https://github.com/totherme/kubectl/blob/6c4975851eabe176923b978af0ca77097305b3f5/pkg/framework/test/apiserver.go#L41-L51) and [fakeapiserver/apiserver.go](https://github.com/totherme/kubectl/blob/6c4975851eabe176923b978af0ca77097305b3f5/pkg/framework/test/assets/fakeapiserver/apiserver.go#L12-L23). We totally agree that this is awkward.

That fake is only used by this one test -- so if we delete this test, we can delete that fake. That might be a good thing -- especially since the wiring which that test was originally written to drive out is now covered by a higher level integration test. The non-happy-path cases in this file cover code that's not covered anywhere else, so we'd definitely like to keep those. 

From a "tests as docs" viewpoint, it feels like a little bit of a pity to leave that file as an incomplete doc -- describing only the error cases... But maybe that's actually just fine? We might have a bit of a think about whether we can write any useful happy-path test (motivated primarily by wanting to keep some executable docs for that path) that doesn't use that fake.